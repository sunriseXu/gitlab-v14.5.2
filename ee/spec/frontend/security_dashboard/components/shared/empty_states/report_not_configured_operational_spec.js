import { shallowMount } from '@vue/test-utils';
import ReportNotConfiguredOperational from 'ee/security_dashboard/components/shared/empty_states/report_not_configured_operational.vue';

describe('Operational report not configured component', () => {
  let wrapper;
  const operationalConfigurationPath = '/operational-configuration';
  const operationalEmptyStateSvgPath = '/operational-placeholder.svg';
  const operationalHelpPath = '/operational-help';

  const createComponent = (dashboardType = 'project') => {
    wrapper = shallowMount(ReportNotConfiguredOperational, {
      provide: {
        dashboardType,
        operationalConfigurationPath,
        operationalEmptyStateSvgPath,
        operationalHelpPath,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('matches the snapshot for projects', () => {
    createComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('matches the snapshot for groups', () => {
    createComponent('group');
    expect(wrapper.element).toMatchSnapshot();
  });

  it('matches the snapshot for instances', () => {
    createComponent('instance');
    expect(wrapper.element).toMatchSnapshot();
  });
});
