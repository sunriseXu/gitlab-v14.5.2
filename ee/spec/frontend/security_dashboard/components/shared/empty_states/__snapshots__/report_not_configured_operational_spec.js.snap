// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operational report not configured component matches the snapshot for groups 1`] = `
<gl-empty-state-stub
  description="Manage and track vulnerabilities identified in your Kubernetes clusters. Vulnerabilities appear here after you create a scan execution policy in any project in this group."
  primarybuttonlink="/operational-configuration"
  primarybuttontext=""
  secondarybuttonlink="/operational-help"
  secondarybuttontext="Learn more"
  svgpath="/operational-placeholder.svg"
  title="Monitor vulnerabilities across clusters"
/>
`;

exports[`Operational report not configured component matches the snapshot for instances 1`] = `
<gl-empty-state-stub
  description="Manage and track vulnerabilities identified in your Kubernetes clusters. Vulnerabilities appear here after you create a scan execution policy in any project in this instance."
  primarybuttonlink="/operational-configuration"
  primarybuttontext=""
  secondarybuttonlink="/operational-help"
  secondarybuttontext="Learn more"
  svgpath="/operational-placeholder.svg"
  title="Monitor vulnerabilities across clusters"
/>
`;

exports[`Operational report not configured component matches the snapshot for projects 1`] = `
<gl-empty-state-stub
  description="Manage and track vulnerabilities identified in your Kubernetes clusters. Vulnerabilities appear here after you create a scan execution policy in this project."
  primarybuttonlink="/operational-configuration"
  primarybuttontext="Create policy"
  secondarybuttonlink="/operational-help"
  secondarybuttontext="Learn more"
  svgpath="/operational-placeholder.svg"
  title="Monitor vulnerabilities across clusters"
/>
`;
